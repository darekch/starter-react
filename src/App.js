import React, { Component } from 'react';
import { hot } from 'react-hot-loader/root';

import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Hello World</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
      </div>
    );
  }
}

export default hot(App);
