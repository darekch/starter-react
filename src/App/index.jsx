import React from 'react';
import { ConnectedRouter } from 'connected-react-router';
import { Provider } from 'react-redux';

import Routes from './routes';
import history from './store/history';
import configureStore from './store/configureStore';

const store = configureStore();

const App = () => (
  <Provider store={store}>
    <ConnectedRouter history={history}>
      <Routes />
    </ConnectedRouter>
  </Provider>
);

export default App;
