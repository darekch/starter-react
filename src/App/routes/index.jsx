import React from 'react';
import { Route, Switch, Link } from 'react-router-dom';

import RoutesErrorBoundary from 'common/errors/boundaries/routes';
// import Items from 'features/Items/containers/ItemsList/ItemsList';
import ItemsContainers from 'features/Items';

const routes = () => (
  <div>
    <nav>
      <Link to="/">Home</Link>
      <Link to="/items">Items</Link>
    </nav>
    <Switch>
      <RoutesErrorBoundary>
        <Route exact path="/items" component={ItemsContainers.ItemsListContainer} />
      </RoutesErrorBoundary>
    </Switch>
  </div>
);

export default routes;
