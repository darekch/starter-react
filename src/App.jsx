import React, { Component } from 'react';
import { hot } from 'react-hot-loader/root';

import Button from 'components/Button';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Hello World</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
        <Button />
      </div>
    );
  }
}

export default hot(App);
