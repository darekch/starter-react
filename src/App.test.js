import React from 'react';
import ReactDOM from 'react-dom';

import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import App from './App';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('should render App', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('p').exists()).toBe(true);
  });

  it('snapshot', () => {
    const tree = mount(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
