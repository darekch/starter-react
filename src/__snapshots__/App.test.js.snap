// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <h1>
      Hello World
    </h1>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </p>
    <Button>
      <button
        type="submit"
      >
        Button
      </button>
    </Button>
  </div>
</App>
`;
